### PR TITLE
[a-g]: replace plist with service block

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/activemq.rb [hunk 1]
```diff
@@ -20,31 +20,9 @@ def install
...
-  def plist
-    <<~EOS
-      <?xml version="1.0" encoding="UTF-8"?>
-      <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN"
-      "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-      <plist version="1.0">
-        <dict>
-          <key>Label</key>
-          <string>#{plist_name}</string>
-          <key>RunAtLoad</key>
```
Comment (carlocab) [inline Formula/activemq.rb:34]:
```
run_type :immediate
```
?

File: Formula/apache-archiva.rb [hunk 2]
```diff
@@ -23,37 +23,10 @@ def post_install
...
-          <key>Label</key>
-          <string>#{plist_name}</string>
-          <key>ProgramArguments</key>
-          <array>
-            <string>#{opt_bin}/archiva</string>
-            <string>console</string>
-          </array>
-          <key>Disabled</key>
-          <false/>
-          <key>RunAtLoad</key>
```
Comment (carlocab) [inline Formula/apache-archiva.rb:43]:
```ruby
run_type :immediate
```
?

File: Formula/bind.rb [hunk 3]
```diff
@@ -103,30 +103,8 @@ def named_conf
...
-          <string>-L</string>
-          <string>#{var}/log/named/named.log</string>
-        </array>
-        <key>ServiceIPC</key>
-        <false/>
-      </dict>
-      </plist>
-    EOS
+  service do
+    run opt_sbin/"named", "-f", "-L", var/"log/named/named.log"
```
Comment (SMillerDev) [inline Formula/bind.rb:107]:
```suggestion
    run [opt_sbin/"named", "-f", "-L", var/"log/named/named.log"]
```

File: Formula/distcc.rb [hunk 4]
```diff
@@ -48,32 +48,10 @@ def install
...
-              <string>--no-detach</string>
-              <string>--allow=192.168.0.1/24</string>
-          </array>
-          <key>WorkingDirectory</key>
-          <string>#{opt_prefix}</string>
-        </dict>
-      </plist>
-    EOS
+  service do
+    run opt_bin/"distcc", "--allow=192.168.0.1/24"
```
Comment (SMillerDev) [inline Formula/distcc.rb:52]:
```suggestion
    run [opt_bin/"distcc", "--allow=192.168.0.1/24"]
```

**Comment by SMillerDev** [comment]
```
==> dig @127.0.0.1 -p 61416 example.com.
dyld: Library not loaded: /opt/homebrew/opt/json-c/lib/libjson-c.5.dylib
  Referenced from: /opt/homebrew/bin/dig
  Reason: image not found
```

and a checksum mismatch for `ejabberd`, not bad

File: Formula/dnsmasq.rb [hunk 5]
```diff
@@ -60,30 +60,9 @@ def post_install
...
-          </array>
-          <key>RunAtLoad</key>
-          <true/>
-          <key>KeepAlive</key>
-          <true/>
-        </dict>
-      </plist>
-    EOS
+  service do
+    run [opt_bin/"dnsmasq", "--keep-in-foreground", "-C", etc/"dnsmasq.conf", "-7", etc/"dnsmasq.d,*.conf"]
```
Comment (SMillerDev) [inline Formula/dnsmasq.rb:64]:
```suggestion
    run [opt_sbin/"dnsmasq", "--keep-in-foreground", "-C", etc/"dnsmasq.conf", "-7", etc/"dnsmasq.d,*.conf"]
```

**Comment by SMillerDev** [comment]
I don't think any of the errors are related to the service blocks. @Homebrew/core opinions?

**Comment by cho-m** [comment]
- `coredns` (All)
- `carrot2` (Big Sur/Catalina)
- `ejabberd` (All) - `Error: SHA256 mismatch`
- `fail2ban` (Intel) - Unrelated missing build dependency in `sphinx-doc`.
- `lanraragi` (ARM) - Looks like common failure seen in other PRs.
- `cntlm` (Mojave) - Temporary network issue

**Comment by SMillerDev** [comment]
I dropped the formulae that were having issues. We can always revisit those later (when cron support is added and we can drop the Plists completely)

**Comment by carlocab** [comment]
`lanraragi` failed on ARM, but that test is flaky (despite my attempts to fix it...). We can probably ignore it.

**Comment by SMillerDev** [comment]
Fails only on ubuntu due to missing dependencies. @carlocab good to merge?

**Comment by carlocab** [comment]
Do we need new bottles for this? If yes, unfortunately not: the ARM runner died 19m into the CI run for some reason. If we don't, then I'm fine with merging this.

**Comment by SMillerDev** [comment]
Nope, it's all in postinstall.

**Review by carlocab — APPROVED** [review]

**Pull request merged**